### ISSUE TITLE
Sort parameters by status?

### ISSUE THREAD

**Issue by esaracco** [apidoc/apidoc#1020]
For now, the parameters are sorted as provided by the developer in its code documentation.
But it would be nice if they were sorted according to their status (`required` / `optional`).
Does it make sens? if so, I can do a PR.

**Comment by NicolasCARPi** [comment]
Let's finish with the first PR first. In the past I made the mistake of merging relatively bad contributions and I felt it strongly when I went deep into the codebase. Now I'm very careful about adding code as I'll be the one maintaining it down the road.

Also, to be honest, if you want to contribute in a meaningful way to this software, what would really help is to fix the version compare code (starting with [this commented line](https://github.com/apidoc/apidoc/blob/6c881ca83e81acc623f7911efaf6d8ef438e247d/template/src/main.js#L27) and [this kind of code](https://github.com/apidoc/apidoc/blob/fdcc9bb4dad445667edc4733f32ccc8bf5abe69f/template/src/hb_helpers.js#L268-L275).

You see, I recently rewrote nearly the entire app and made a lot of major changes to the code (like removing require.js and using webpack, using es6 classes, etc...). I got everything fixed again and the app is running nicely for most projects. Except for this part.

And this broken part is the last thing stopping me from releasing 0.50.0 into the wild. So before adding more code, the priority is to fix existing bugs or regressions.

Yes it does make sense to sort them BUT be very careful as sorting things have been tried and it turned out badly, see https://github.com/apidoc/apidoc/issues/968#issuecomment-869046654 where in the end I just removed the code sorting things and everyone was happy ever after (and that code has been "fixed" multiple times before by different persons). Also this is something that one can easily achieve by adding the required params first in the code (and no code to maintain on my side!).

So this is probably not trivial to implement correctly and I think it's best if you focus on some other issues, like the versioning that I mentioned ;)

Cheers,
~Nico
